### PR TITLE
Git ignore hidden files with defined exceptions

### PR DESCRIPTION
**Pull request by mosfet80**
Thank you for creating a pull request to contribute to FreeCAD! To ease integration, we ask you to conform to the following items. Pull requests which don't satisfy all the items below might be rejected. If you are in doubt with any of the items below, don't hesitate to ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10)!

- [ ]  Your pull request is confined strictly to a single module. That is, all the files changed by your pull request are either in `App`, `Base`, `Gui` or one of the `Mod` subfolders. If you need to make changes in several locations, make several pull requests and wait for the first one to be merged before submitting the next ones
- [ ]  In case your pull request does more than just fixing small bugs, make sure you discussed your ideas with other developers on the FreeCAD forum
- [x]  Your branch is [rebased](https://git-scm.com/docs/git-rebase) on latest master `git pull --rebase upstream master`
- [ ]  All FreeCAD unit tests are confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [ ]  All commit messages are [well-written](https://chris.beams.io/posts/git-commit/) ex: `Fixes typo in Draft Move command text`
- [ ]  Your pull request is well written and has a good description, and its title starts with the module name, ex: `Draft: Fixed typos`
- [ ]  Commit messages include `issue #<id>` or `fixes #<id>` where `<id>` is the issue ID number from our [Issues database](https://github.com/FreeCAD/FreeCAD/issues) in case a particular commit solves or is related to an existing issue. Ex: `Draft: fix typos - fixes #4805`

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.20 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=56135).

---

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # file types to ignore
-
+.*
+!/.gitignore
+!/.travis/
+!/.travis.yml
```
Comment (adrianinsaval) [inline .gitignore:5]: is travis even used anymore?

Comment (mosfet80) [reply .gitignore:5]: there is the .travis folder, Seeing its contents, for me it can be removed

**Comment by berndhahnebach** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7108">pipeline status</a> for feature branch [PR_7108](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7108). Pipeline [578985894](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/578985894) was triggered at [e7ede5a](https://github.com/FreeCAD/FreeCAD/pull/7108/commits/e7ede5a0348c5810c1a834f94cab0bf813d084b9). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

File: .gitignore (hunk 1 above)
Comment (donovaly) [reply .gitignore:5]: Let's keep it for now.

I will make a separate PR to remove .travis and see what he developers think about it.

Comment (donovaly) [reply .gitignore:5]: Here it is: #7238

**Pull request merged**